### DISCUSSION TITLE
Image upload in post/page options

### DISCUSSION THREAD

**Discussion by gpsblues** [GetPublii/Publii#1664] [Help] [answered]
Hi,
among the new features of P.46 I understood that the "image support" is integrated in the configuration options:

```
"postConfig": [
        {
            "name": "upload",
            "label": "Upload",
            "note": "Upload control",
            "value": "",
            "type": "upload"
        },
        {
            "name": "file",
            "label": "Small upload",
            "note": "Small upload control",
            "value": "",
            "type": "smallupload"
        },
        ...
]
```

However, I can't get the desired result. Theme Simple v3.
Can you tell me if I forgot or did something wrong?

![image](https://github.com/user-attachments/assets/30c59a9d-a981-421f-9f2f-43b94c40115d)

**Comment by bobmitro** [comment, marked as answer]
The **upload** control is dedicated to uploading files and doesn't work in the post/pageConfig options. Instead, use the **image** type.

```
{
            "name": "upload",
            "label": "Upload",
            "note": "Upload control",
            "value": "",
            "type": "image"
        },
```

More about it: https://getpublii.com/dev/post-config-options/#image-upload

**Comment by gpsblues** [comment]
Thanks a lot Bob,

I didn't understand, sorry. This is a great feature.
It would be nice to have also smallimageupload with which maybe to upload a preview video... :)
